sdks/python/container/base_image_requirements_manual.txt

```diff
@@ -38,4 +38,7 @@ nose==1.3.7  # For Dataflow internal testing. TODO: remove this.
 python-snappy  # Optimizes execution of some Beam codepaths.
 scipy
 scikit-learn
-build>=1.0,<2 # tool to build sdist from setup.py in stager.
\ No newline at end of file
+build>=1.0,<2 # tool to build sdist from setup.py in stager.
+# Authentication for Google Artifact Registry when using --extra-index-url or --index-url in requirements.txt in
+# Dataflow, which allows installing python packages from private Python repositories in GAR.
+keyrings.google-artifactregistry-auth
```

Comment:
I'm good with the change, just note another choice will be add this dependency to [gcp] extension in setup.py. Is there a preference here?

Reply:
+1 - I think adding it to the gcp extension ([here](https://github.com/apache/beam/blob/a7852d9ddb1726838c5878601c582fae45a8798b/sdks/python/setup.py#L428)) is probably a slightly better option.

Reply:
Thanks! I moved it to [gcp] extension in setup.py.